drivers/sx127x/sx127x_netdev.c

```diff
@@ -0,0 +1,499 @@
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_sx127x
+ * @{
+ * @file
+ * @brief       Netdev adaptation for the sx127x driver
+ *
+ * @author      Eugene P. <ep@unwds.com>
+ * @author      José Ignacio Alamos <jose.alamos@inria.cl>
+ * @}
+ */
+
+#include <stddef.h>
+#include <string.h>
+#include <errno.h>
+
+#include "net/netopt.h"
+#include "net/netdev.h"
+#include "sx127x_registers.h"
+#include "sx127x_internal.h"
+#include "sx127x_netdev.h"
+#include "sx127x.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+/* Internal helper functions */
+static uint8_t _get_tx_len(const struct iovec *vector, unsigned count);
+static int _set_state(sx127x_t *dev, netopt_state_t state);
+static int _get_state(sx127x_t *dev, void *val);
+
+/* Netdev driver api functions */
+static int _send(netdev_t *netdev, const struct iovec *vector, unsigned count);
+static int _recv(netdev_t *netdev, void *buf, size_t len, void *info);
+static int _init(netdev_t *netdev);
+static void _isr(netdev_t *netdev);
+static int _get(netdev_t *netdev, netopt_t opt, void *val, size_t max_len);
+static int _set(netdev_t *netdev, netopt_t opt, void *val, size_t len);
+
+const netdev_driver_t sx127x_driver = {
+    .send = _send,
+    .recv = _recv,
+    .init = _init,
+    .isr = _isr,
+    .get = _get,
+    .set = _set,
+};
+
+static int _send(netdev_t *netdev, const struct iovec *vector, unsigned count)
+{
+    sx127x_t *dev = (sx127x_t*) netdev;
+    uint8_t size;
+    size = _get_tx_len(vector, count);
+    switch (dev->settings.modem) {
+        case SX127X_MODEM_FSK:
+            /* todo */
+            break;
+        case SX127X_MODEM_LORA:
+            if (dev->settings.lora.iq_inverted) {
+                sx127x_reg_write(dev, SX127X_REG_LR_INVERTIQ,
+                                 ((sx127x_reg_read(dev, SX127X_REG_LR_INVERTIQ) &
+                                   SX127X_RF_LORA_INVERTIQ_TX_MASK & SX127X_RF_LORA_INVERTIQ_RX_MASK) |
+                                  SX127X_RF_LORA_INVERTIQ_RX_OFF | SX127X_RF_LORA_INVERTIQ_TX_ON));
+                sx127x_reg_write(dev, SX127X_REG_LR_INVERTIQ2,
+                                 SX127X_RF_LORA_INVERTIQ2_ON);
+            }
+            else {
+                sx127x_reg_write(dev, SX127X_REG_LR_INVERTIQ,
+                                 ((sx127x_reg_read(dev, SX127X_REG_LR_INVERTIQ) &
+                                   SX127X_RF_LORA_INVERTIQ_TX_MASK & SX127X_RF_LORA_INVERTIQ_RX_MASK) |
+                                  SX127X_RF_LORA_INVERTIQ_RX_OFF | SX127X_RF_LORA_INVERTIQ_TX_OFF));
+                sx127x_reg_write(dev, SX127X_REG_LR_INVERTIQ2, SX127X_RF_LORA_INVERTIQ2_OFF);
+            }
+
+            /* Initializes the payload size */
+            sx127x_reg_write(dev, SX127X_REG_LR_PAYLOADLENGTH, size);
```

Comment:
Also, you have a [`sx127x_set_payload_length()`](https://github.com/RIOT-OS/RIOT/pull/6797/files#diff-dada00589373226016632c37cf27ea59R671) function implemented that does effectively the same. So why not use it